setup.py

```diff
@@ -69,3 +56,5 @@ def _get_version() -> str:
     python_requires='>=3.5',
     install_requires=["PyYAML"],
 )
+
+version_backup.rename(version_path)
```

Comment:
Is that a sane thing to do? move version to a backup file, run setup, and then move it *back* into place?
Can we get a similar behavior without the rename magic?
Especially the fact that we are mutating the source underneath you and then pretending it never happened feels racy to me. (your editor is going to complain, some other process happening concurrently may or may not see what you think it will, etc.)

Reply:
>  Can we get a similar behavior without the rename magic?

probably, by subclassing the setup commands and adding the behaviour in there (for every command we care about). There doesn't seem to be a 'pre-build' hook (but i might be wrong).

it'd be a lot more code and complexity i fear

> Especially the fact that we are mutating the source underneath you and then pretending it never happened feels racy to me. (your editor is going to complain, some other process happening concurrently may or may not see what you think it will, etc.)

anything running on the codebase concurrently with a release is bad news
your editor should only complain if it has unsaved changes to a file (i know mine does that :) )